server/supernova/server/main.cpp

```diff
@@ -335,24 +335,33 @@ int main(int argc, char* argv[]) {
     cout << "compiled for debugging" << endl;
 #endif
 
-    server_shared_memory_creator::cleanup(args.port());
-    nova_server server(args);
-    register_signal_handler();
-
-    set_plugin_paths(args, sc_factory.get());
-    load_synthdefs(server, args);
-
-    if (!args.non_rt) {
-        try {
-            start_audio_backend(args);
-            cout << "Supernova ready" << endl;
-        } catch (exception const& e) {
-            cout << "Error: " << e.what() << endl;
-            exit(1);
-        }
-        server.run();
-    } else
-        server.run_nonrt_synthesis(args);
+    // FIXME should have more granular error handling
+    try {
+        server_shared_memory_creator::cleanup(args.port());
+        nova_server server(args);
+        register_signal_handler();
+
+        set_plugin_paths(args, sc_factory.get());
+        load_synthdefs(server, args);
+
+        if (!args.non_rt) {
+            try {
+                start_audio_backend(args);
+                cout << "Supernova ready" << endl;
+            } catch (exception const& e) {
+                cout << "\n*** Error starting audio backend: " << e.what() << endl;
+                exit(1);
+            }
+            server.run();
+        } else
+            server.run_nonrt_synthesis(args);
+    } catch (const std::exception& exc) {
+        cout << "\n*** Error in main(): " << exc.what() << endl;
+        exit(1);
+    } catch (...) {
+        cout << "\n*** Unknown error in main()" << endl;
+        exit(1);
+    }
```

Comment:
this is not great, but still a strict improvement over the old behavior, which was to let exceptions bubble out of main to be caught by the language runtime. of course, fixing error handling for the whole project would be out of scope for this fix